sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1060,53 +1060,36 @@ public void populateDisplayData(DisplayData.Builder builder) {
           getUpperBound() - getLowerBound() >= getNumPartitions(),
           "The specified number of partitions is more than the difference between upper bound and lower bound");
 
-      if (getUpperBound() == MAX_VALUE || getLowerBound() == 0) {
```

Comment:
in this case, the default values will be used. We could change the parameter to be `Long` instead of `long` and fail with nulls. WDYT?

Reply:
Sounds good to me. I think it would be useful to add an assert that it requires to specify the upper and lower bounds if `readWithPartitions()` is used.

Reply:
done